### PR TITLE
[4.0]Replace removed constant RELEASE with correct code, fix #18154

### LINKED ISSUE #18154: [4.0] - install Undefined class constant 'RELEASE'
### Steps to reproduce the issue
install 4.0 staging

### Expected result
all ok 

### Actual result
screenshot from 2017-09-28 21-11-26

### PR DESCRIPTION
**Pull request by joomdonation**
Pull Request for Issue # .

### Summary of Changes
This small PR replace the removed RELEASE  constant with correct code, fix issue #18154 while installing Joomla 4.0

### Testing Instructions
Try to install latest Joomla 4.0-dev. Before patch, there is error and install fails. After patch, install success

### CODE REVIEW

**Comment by zero-24** [comment]
should't this better be patched in 3.x and than merged into 4.0?

**Comment by zero-24** [comment]
https://github.com/joomla/joomla-cms/blob/staging/libraries/src/Version.php#L264

File: libraries/src/Version.php [hunk 1]
```diff
@@ -213,11 +213,11 @@ public function getUserAgent($component = null, $mask = false, $addVersion = tru
 
 		if ($addVersion)
 		{
-			$component .= '/' . self::RELEASE;
+			$component .= '/' . self::MAJOR_VERSION . '.' . self::MINOR_VERSION;
 		}
 
 		// If masked pretend to look like Mozilla 5.0 but still identify ourselves.
-		return ($mask ? 'Mozilla/5.0 ' : '') . self::PRODUCT . '/' . self::RELEASE . '.' . self::DEV_LEVEL . ($component ? ' ' . $component : '');
+		return ($mask ? 'Mozilla/5.0 ' : '') . self::PRODUCT . '/' . self::MAJOR_VERSION . '.' . self::MINOR_VERSION . ($component ? ' ' . $component : '');
```
Comment (mbabker) [inline libraries/src/Version.php]: Err, this change isn't quite returning the same result as before it looks like this should basically be `$this->getShortVersion()`

**Comment by joomdonation** [comment]
Method getUserAgent uses another removed constant DEV_LEVEL which I don't what should be used to replace (or if it is needed), so I remove it for now

**Comment by mbabker** [comment]
The old `RELEASE` constant is a combination of the new`MAJOR_VERSION` and `MINOR_VERSION` constants.

The old `DEV_LEVEL` constant is a combination of the new `PATCH_VERSION` and `EXTRA_VERSION` constants.

For both 3.x and 4.0 https://github.com/joomla/joomla-cms/blob/3.8.0/libraries/src/Version.php#L279 and https://github.com/joomla/joomla-cms/blob/3.8.0/libraries/src/Version.php#L283 should replace the calls for the version number constants to a call to `$this->getShortVersion()` which is essentially the same result.

File: libraries/src/Version.php (hunk 1 above)
Comment (joomdonation) [reply libraries/src/Version.php]: Yes, seems so. I am going to make change as you said

**Comment by zero-24** [comment]
> should't this better be patched in 3.x and than merged into 4.0?

@wilsonge this way your pain in merging 3 to 4 is even bigger ..

**Comment by wilsonge** [comment]
I mean the stuff in version.php can be updated. But the updater code i'd be nervous about updating. I *think* it's ok. But not sure what the deal would be about direct updates

**Pull request merged**